### ISSUE TITLE
openCV debug mode LNK2019 / LNK1120 error

### ISSUE THREAD

**Issue by jagrme00** [microsoft/vcpkg#12487]
Hello,

I'm trying to build a x64-windows project in debug mode but I get the following two errors:

Error | LNK2019 | unresolved external symbol "void __cdecl cv::cvtColor(class cv::_InputArray const &,class cv::_OutputArray const &,int,int)" (?cvtColor@cv@@YAXAEBV_InputArray@1@AEBV_OutputArray@1@HH@Z) referenced in function "public: static void __cdecl IntensityToRGBConverter::WinterColormap(double,double *)" (?WinterColormap@IntensityToRGBConverter@@SAXNPEAN@Z) | AMADEUS | C:\MA_Repo\30_Code\85_Code_Gruenewald\01_RTMaps\MyRTMaps_SDK\AMADEUS.u\Intensity_To_RGB_Converter.obj | 1 |  

Error | LNK1120 | 1 unresolved externals | AMADEUS | C:\MA_Repo\30_Code\85_Code_Gruenewald\01_RTMaps\MyRTMaps_SDK\packages\debug\AMADEUS.pck | 1 |  

The build errors are disappearing when I switch to release mode.

I use vcpkg version f2e9ef7f259843e116453bbabb7bcc2b13246f13, with

opencv4:x64-windows                                4.1.1-3          computer vision library
opencv4[dnn]:x64-windows                                            Enable dnn module
opencv4[jpeg]:x64-windows                                           JPEG support for opencv
opencv4[opengl]:x64-windows                                         opengl support for opencv
opencv4[png]:x64-windows                                            PNG support for opencv
opencv4[tiff]:x64-windows                                           TIFF support for opencv
opencv4[webp]:x64-windows                                           WebP support for opencv
opencv:x64-windows                                 4.1.1-1          Computer vision library
opencv[dnn]:x64-windows                                             Enable dnn module
opencv[jpeg]:x64-windows                                            JPEG support for opencv
opencv[opengl]:x64-windows                                          opengl support for opencv
opencv[png]:x64-windows                                             PNG support for opencv
opencv[tiff]:x64-windows                                            TIFF support for opencv
opencv[webp]:x64-windows                                            WebP support for opencv

Why is this happening?

Thanks in advance!

**Comment by NancyLi1013** [comment]
Hi @jagrme00 
Thanks for posting this issue.
Could you please try to rebuild this based on PR #11130?

**Comment by jagrme00** [comment]
Well I can try it, but this changes my local repo to version dbfe6cc1ca3ce3532b768acd2ccd8c60616a412f. Thus the version of the Point Cloud Library (PCL) that I also need for my project gets changed from 1.9.1-11 to 1.9.1-13 and this will break my code since not all features are included in the current PCL version.

Is it possible to run two separate vcpkg versions in parallel one for PCL and the other for OpenCV to overcome this issue or is there a better solution?

Thanks!

**Comment by NancyLi1013** [comment]
@jagrme00 
You can merge the two changes to one branch and then try to rebuild. Since there are too many changes in PR #11130. You can update the changes for `PCL `to PR #11130.

**Comment by jagrme00** [comment]
Okay. Since I am not working with git on a daily basis can you explain this with a little bit more details please?

I tried the following:

1) git clone https://github.com/Microsoft/vcpkg.git

2) git checkout f2e9ef7f259843e116453bbabb7bcc2b13246f13

3) git fetch origin pull/11130/head:TEST

4) git merge TEST

I am 100% sure that these are the wrong steps because I ended up with PCL 1.9.1-13 again...

**Comment by NancyLi1013** [comment]
@jagrme00 
you might try to use the following steps:

- git clone https://github.com/cenit/vcpkg.git
- cd vcpkg
- git checkout -b dev/cenit/opencv43 origin/dev/cenit/opencv43
- merge PCL updates to this repo manually

Or you can wait for the PR merged into master and then try this again.

**Comment by talregev** [comment]
> @jagrme00
> you might try to use the following steps:
> 
> * git clone https://github.com/cenit/vcpkg.git
> * cd vcpkg
> * git checkout -b dev/cenit/opencv43 origin/dev/cenit/opencv43
> * merge PCL updates to this repo manually
> 
> Or you can wait for the PR merged into master and then try this again.

@NancyLi1013 @jagrme00 to do it with less line:
* git clone https://github.com/cenit/vcpkg.git  -b dev/cenit/opencv43
* cd vcpkg
* merge PCL updates to this repo manually

**Comment by NancyLi1013** [comment]
Hi @jagrme00 
opencv4 has been updated to 4.3 via PR #11130. 

Now you can update vcpkg to the latest version and try to rebuild if this can work for you.

**Comment by NancyLi1013** [comment]
@jagrme00 
Please let me know if this is still a problem for you now.

**Comment by jagrme00** [comment]
@NancyLi1013 sorry for my delay. Currently I have some other topics which require more attention than this issue. 
I'll try the suggestions within the next few days and I'll let you know my results. 

Thanks for your help.

**Comment by jagrme00** [comment]
My problems are solved now. The above mentioned approach https://github.com/microsoft/vcpkg/issues/12487#issuecomment-664813848 is working for me.
Thank you @NancyLi1013 for you help!

**Comment by NancyLi1013** [comment]
Thanks for your update.
